### PR TITLE
Remove adapter walk

### PR DESCRIPTION
**Pull request by nopcoder**
Removing blockstore level Walk functionality. Currently the ingest package is used to walk on underlying storage.
This change is to remove capability duplication and keeps code that used by the project functionality.

    - Remove the Graveler proto definition for `GarbageCollectionRunMetadata`
    - Uncommitted GC information not reuse `GarbageCollectionRunMetadata` -
      set it own type
    - Prepare UGC return for each call the object name created
    - diagnostics command no longer list metaranges and ranges. Leaving
      diagnostics command to do nothing.
    - Catalog test uses the return locaiton and object name to read parquet
      information and match to expected without using Walk.

Part of #5341

### CODE REVIEW

**Review by itaiad200 — APPROVED** [review]
LGTM

File: pkg/catalog/catalog.go [hunk 1]
```diff
@@ -1770,29 +1771,37 @@ type GCUncommittedMark struct {
 }
 
 type PrepareGCUncommittedInfo struct {
-	Mark     *GCUncommittedMark
-	Metadata graveler.GarbageCollectionRunMetadata
+	RunID      string `json:"run_id"`
+	Location   string `json:"location"`
+	ObjectName string `json:"object_name"`
+	Mark       *GCUncommittedMark
```
Comment (N-o-Z) [inline pkg/catalog/catalog.go:1777]: I think these changes are unrelated to the PR and should be done in a different issue

Comment (nopcoder) [reply pkg/catalog/catalog.go:1777]: I've returned the ObjectName as part of the response.
So I've used different return object type.
The object type that was used was defined in proto file - this change I can move to a different PR.
But I wanted to remove the unused field.
So I also moved it out of the proto as it is not serialized as proto.

**Review by N-o-Z — COMMENTED** [review]
Thanks! 
What's troubling me with making the changes on 2 separate issues is the fact that you need to refactor code that uses the walk interface - find a solution that doesn't use it only to bring it back in the following PR.

**Comment by nopcoder** [comment]
> Thanks! What's troubling me with making the changes on 2 separate issues is the fact that you need to refactor code that uses the walk interface - find a solution that doesn't use it only to bring it back in the following PR.

Think of this change as deprecating a single behaviour - didn't want to mix the ingest functionality into block adapter.
Also didn't want to use the ingest solution for diagnostics and test code.

File: cmd/lakefs/cmd/diagnostics.go [hunk 2]
```diff
@@ -2,69 +2,16 @@ package cmd
...
-	"github.com/treeverse/lakefs/pkg/block/factory"
-	"github.com/treeverse/lakefs/pkg/catalog"
-	"github.com/treeverse/lakefs/pkg/diagnostics"
-	"github.com/treeverse/lakefs/pkg/kv"
-	"github.com/treeverse/lakefs/pkg/stats"
-	"github.com/treeverse/lakefs/pkg/upload"
 )
 
-// diagnosticsCmd represents the diagnostics command
 var diagnosticsCmd = &cobra.Command{
```
Comment (johnnyaug) [inline cmd/lakefs/cmd/diagnostics.go:9]: Why not totally remove this command?

File: pkg/catalog/catalog.go [hunk 3]
```diff
@@ -1770,29 +1771,37 @@ type GCUncommittedMark struct {
 }
 
 type PrepareGCUncommittedInfo struct {
-	Mark     *GCUncommittedMark
-	Metadata graveler.GarbageCollectionRunMetadata
+	RunID      string `json:"run_id"`
+	Location   string `json:"location"`
+	ObjectName string `json:"object_name"`
```
Comment (johnnyaug) [inline pkg/catalog/catalog.go:1776]:
```suggestion
	BasePath   string `json:"base_path"`
	Filename string `json:"filename"`
```
"object" is more correct, but less comprehensible.

File: cmd/lakefs/cmd/diagnostics.go (hunk 2 above)
Comment (nopcoder) [reply cmd/lakefs/cmd/diagnostics.go:9]: Thought we would like to use it as some point - but I agree that it is strange to keep something without any functionality.

File: pkg/catalog/catalog.go (hunk 3 above)
Comment (nopcoder) [reply pkg/catalog/catalog.go:1776]: You update location to base_path - it is the same as gc metadata parameters. Do you still think we need to break alignment/

Comment (johnnyaug) [reply pkg/catalog/catalog.go]: Not really - the object name was mostly what bothered me.

**Review by johnnyaug — APPROVED** [review]
